backend/dataline/services/llm_flow/toolkit.py

```diff
@@ -64,14 +59,13 @@ def execute_sql_query(
     db: SQLDatabase, query: str, for_chart: bool = False, chart_type: Optional[ChartType] = None
 ) -> QueryRunData:
     """Execute the SQL query and return the results or an error message."""
-    result = cast(Result[Any], db.run(query, fetch="cursor", include_columns=True))  # type: ignore[misc]
+    columns, rows = db.custom_run_sql(query)
     truncated_rows = []
-    for row in result.fetchall():
+    for row in rows:
```

Comment:
Same comment here - why not keep the same API?

Reply:
We can't do `result.fetchall` if the dialect is mssql. The connection of the cursor must not be closed for it to run successfully